### DISCUSSION TITLE
Dataset formats supported in YOLOv5 and YOLOv8 for classification, detection, segmentation

### DISCUSSION THREAD

**Discussion by ecarrenolozano** [ultralytics/ultralytics#708] [Q&A] [answered]
Good afternoon dear community,

I am looking for information related to the different dataset formats supported for YOLOv5 and YOLOv8 models. In the documentation, I found the following:
 
**Detection:**
  - normalized xywh

_Please only reply to this discussion thread in an organized way in order to help and contribute to the general documentation of the YOLO project. For instance:_

**Detection formats:**
YOLOv5 format [1]
 -  One row per object.
 - Each row is `class x_center y_center width height` format.
 - Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and width by image `width`, and `y_center` and `height` by image height.
  -Class numbers are zero-indexed (start from 0).

References: 
[1] [Train custom data.](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)

**Comment by glenn-jocher** [comment, marked as answer]
@ecarrenolozano good afternoon!

For YOLOv8, the dataset formats supported for various computer vision tasks are as follows:

**Detection:**
- YOLOv8 expects the dataset in a similar format as YOLOv5, with one row per object and each row containing `class x_center y_center width height` in normalized xywh format. The class indices are zero-indexed.

**Segmentation:**
- YOLOv8 supports segmentation tasks, and the dataset should include images along with corresponding segmentation masks. The masks should be in a format that delineates different regions of interest.

**Classification:**
- For image classification, YOLOv8 requires a dataset with labeled images. The labels should correspond to the class categories the model needs to learn.

**Pose/Keypoint Estimation:**
- Datasets for pose estimation should include images and corresponding keypoints annotations. Each keypoint is typically represented by its coordinates and a visibility flag.

For detailed instructions on how to prepare your datasets for these tasks, please refer to the Ultralytics Docs. The documentation provides comprehensive guidance on structuring your data correctly for training with YOLOv8.

Remember, the success of your model depends greatly on the quality and format of your dataset, so it's crucial to follow the guidelines closely. Happy training! 😊

**Reply by ecarrenolozano** [reply]
Hi @glenn-jocher good morning,

Thank you for the detailed explanation here.

**Reply by glenn-jocher** [reply]
@ecarrenolozano good morning!

You're welcome! If you have any more questions or need further assistance with YOLOv8 or any other Ultralytics projects, feel free to reach out. The community and the Ultralytics team are here to help. Have a great day and best of luck with your projects! 😄🚀
